src/richie/apps/search/indexers/courses.py

```diff
@@ -69,8 +70,17 @@ class CoursesIndexer:
     form = CourseSearchForm
 
     scripts = {
+        # We use the same `state` script for several use cases as the code is mostly common and
```

Comment:
You have probably thought of this but just in case

To me those scripts look like python strings. So maybe you could simply concatenate the common part with the 2 different parts that handle the return values to make 2 different scripts without repeating the common part?

Reply:
Yes I thought of it but it would make the script very difficult to read/understand? The performance loss appears negligeable (just one more if... a question of nano seconds?).

Reply:
I made the suggestion more as a way to make this easier to understand than out of performance considerations.

Anyway, there's no clean answer, it was just a suggestion, you can leave it as it is 👍

Reply:
I would cut the algo in 2 different places and make it harder to follow IMO.